internal/common/shared/resources.go

```diff
@@ -0,0 +1,67 @@
+package shared
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/SOMAS2020/SOMAS2020/pkg/miscutils"
+)
+
+// ResourceDistributionStrategy characterises the different ways in which resources can be distributed
+// amongst concerned parties. This is required in splitting foraging returns, for example.
+type ResourceDistributionStrategy int
+
+const (
+	// EqualSplit ignores input contributions
+	EqualSplit ResourceDistributionStrategy = iota
+	// InputProportionalSplit splits proportional to input contributions
+	InputProportionalSplit
+	// RankProportionalSplit splits according to some other external rank specification
+	RankProportionalSplit
+
+	// don't modify this
+	_end
```

Comment:
I usually prefer adding the name of the type--this is to prevent clashes between ends of different iotas.

Reply:
Fair enough - was a bit silly of me